client/dashboard/me/profile-personal-details/index.tsx

```diff
@@ -1,61 +1,133 @@
-import { userSettingsMutation } from '@automattic/api-queries';
-import { useMutation } from '@tanstack/react-query';
+import {
+	isAutomatticianQuery,
+	userSettingsMutation,
+	userSettingsQuery,
+} from '@automattic/api-queries';
+import { useMutation, useSuspenseQuery } from '@tanstack/react-query';
 import {
 	Button,
 	Card,
 	CardBody,
 	CheckboxControl,
-	Notice,
 	__experimentalHStack as HStack,
 	__experimentalVStack as VStack,
 	__experimentalInputControl as InputControl,
 } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
 import { DataForm } from '@wordpress/dataviews';
-import { useMemo } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { useState } from 'react';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect, useMemo, useCallback } from 'react';
 import { SectionHeader } from '../../components/section-header';
+import UsernameSection from './username-section';
 import type { UserSettings } from '@automattic/api-core';
 import type { Field, Form } from '@wordpress/dataviews';
+import './style.scss';
 
-const fields: Field< UserSettings >[] = [
-	{
-		id: 'first_name',
-		label: __( 'First name' ),
-		type: 'text',
-	},
-	{
-		id: 'last_name',
-		label: __( 'Last name' ),
-		type: 'text',
-	},
-	{
-		id: 'user_login',
-		label: __( 'Username' ),
-		type: 'text',
-		Edit: ( { field, data, hideLabelFromVision } ) => {
-			const { getValue } = field;
-			return (
-				<InputControl
-					__next40pxDefaultSize
-					label={ hideLabelFromVision ? '' : field.label }
-					value={ getValue( { item: data } ) }
-					disabled
-					onChange={ () => {} }
-				/>
-			);
+interface PersonalDetailsSectionProps {
+	profile: UserSettings;
+}
+
+export default function PersonalDetailsSection( {
+	profile: serverProfile,
+}: PersonalDetailsSectionProps ) {
+	const { data: userSettings } = useSuspenseQuery( userSettingsQuery() );
+	const { data: isAutomattician } = useSuspenseQuery( isAutomatticianQuery() );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const [ edits, setEdits ] = useState< Partial< UserSettings > >( {} );
+
+	const mutation = useMutation( {
+		...userSettingsMutation(),
+		meta: {
+			snackbar: {
+				success: __( 'Settings saved.' ),
+				error: __( 'Failed to save settings.' ),
+				speak: true,
+				successPoliteness: 'polite',
+				errorPoliteness: 'assertive',
+			},
+		},
+	} );
+
+	const data = useMemo( () => ( { ...serverProfile, ...edits } ), [ serverProfile, edits ] );
+
+	const currentUsername = userSettings?.user_login || '';
+	const isEmailVerified = userSettings?.email_verified ?? true;
+	const canChangeUsername = userSettings?.user_login_can_be_changed ?? true;
+
+	useEffect( () => {
+		const params = new URLSearchParams( window.location.search );
+		if ( params.get( 'usernameChangeSuccess' ) === 'true' ) {
```

Comment:
So great that we're thinking about `aria-live`! It's great that HD is going to have a much better a11y story than v1 Calypso.

For this particular "show message on load" behaviour we already have a `<FlashMessage>` component. Can we bake the `aria-live` functionality into `FlashMessage` instead of implementing it in this one component?

Reply:
> `key` is a [reserved prop](https://reactjs.org/link/special-props) that we can't use so I changed it to `queryParam`

Yes that  `key` prop was an oversight on my part! It was also fixed in #105695, along with other `<FlashMessage>` improvements.